src/vm/gchelpers.cpp

```diff
@@ -1314,7 +1314,7 @@ void ErectWriteBarrierForMT(MethodTable **dst, MethodTable *ref)
 
     *dst = ref;
 
-#ifdef _DEBUG
+#ifdef WRITE_BARRIER_CHECK
```

Comment:
WRITE_BARRIER_CHECK is NOT defined for internal checked builds and that is the reason I didn't define it for checked builds here.